node/overseer/src/lib.rs

```diff
@@ -302,7 +324,83 @@ struct SubsystemInstance<M> {
 #[derive(Debug)]
 pub struct OverseerSubsystemContext<M>{
 	rx: mpsc::Receiver<FromOverseer<M>>,
-	tx: mpsc::Sender<ToOverseer>,
+	tx: mpsc::Sender<MaybeTimed<ToOverseer>>,
+	metrics: Metrics,
+	capture_each: u32,
+	capture_state: u32,
+}
+
+impl<M> OverseerSubsystemContext<M> {
+	/// Create a new `OverseerSubsystemContext`.
+	///
+	/// `capture_each` determines which messages are timed, i.e. a value of 256 means that every 256th
```

Comment:
This is not a good statistical approach. In the good case (which is the case we care about most for timing I assume) a lot of messages have deterministic sequences / follow up messages. 

As such the pick of message should have a true statistical property, such as using a very cheap hash function on `MaybeTimer` creation, i.e. Marsaglia's Xorshift.

At the very least, we should pick a prime for which messages to pick.

Reply:
Agree about the statistics; restored a very simple RNG implementation in ea59309.